Automattic/jetpack
# Donations block: Make currency and amounts editable

Part of https://github.com/Automattic/wp-calypso/issues/42856

#### Changes proposed in this Pull Request:
Adds the ability to edit the currency and amounts in a donations block.

![Jul-30-2020 16-31-39](https://user-images.githubusercontent.com/1233880/88935664-45643d00-d282-11ea-8aa3-08d337fb3bde.gif)

#### Jetpack product discussion
pbMlHh-dW-p2

#### Does this pull request change what data or activity we track or use?
No.

#### Testing instructions:
* With a WordPress instance of your choice, install Jetpack Beta and activate this branch. Or visit https://jurassic.ninja/create/?jetpack-beta&branch=update/donation-block-edit-amount.
* Connect Jetpack to WordPress.com with a paid plan.
* Make sure Jetpack beta blocks are visible by going to /wp-admin/options-general.php?page=companion_settings and activating `JETPACK_BETA_BLOCKS`. 
* Note that if folks are using the sandbox store method (PCYsg-lW4-p2) to also set the `JETPACK__SANDBOX_DOMAIN` value.
* Go to Posts > New and insert a Donations block.
* Make sure the block toolbar displays the current currency. Clicking on it should display a dropdown for changing it.
* Try changing the currency.
* Make sure the amounts automatically change to new defaults based on the [minimum amount valid for a transaction](https://github.com/Automattic/jetpack/blob/0fa76778a8724c55896fd513de2ceabb6877c3a7/extensions/shared/currencies.js#L11-L29) as follows:
  * 1st tier: Minimum amount x10 (e.g. USD 5).
  * 2nd tier: Minimum amount x30 (e.g. USD 15).
  * 3rd tier: Minimum amount x200 (e.g. USD 100).
* Make sure you can edit the amounts by clicking in any of them.
* Try inserting an invalid value for an amount (e.g. a value that doesn't contain any number or a value below the minimum amount). Invalid characters should be removed.
* Save the post and reload the editor.
* Make sure the new amount is preserved.

#### Proposed changelog entry for your changes:
N/A
